### PR TITLE
Convert ModalCurtain to TypeScript

### PR DESCRIPTION
**Pull request by lavelle**

### CODE REVIEW

File: packages/thumbprint-react/components/ModalCurtain/test.tsx [hunk 1]
```diff
@@ -47,23 +47,23 @@ describe('ModalCurtain', () => {
 
         mount(
             <ModalCurtain onCloseClick={jest.fn}>
-                {args => {
+                {(args): JSX.Element => {
                     obj = args;
```
Comment (danoc) [inline packages/thumbprint-react/components/ModalCurtain/test.tsx]: What's the change here? I don't think I understand it.

Comment (lavelle) [reply packages/thumbprint-react/components/ModalCurtain/test.tsx]: not sure what you're asking about specifically here, i just changed it to make the type checking pass. the `children` prop of `ModalCurtain` is a function that returns `JSX.Element`, but `null` is not included in that, so I switched it to an empty `div`

**Comment by danoc** [comment]
This one needs also needs the base branch to be switched to `master`. There were a few conflicts when I tried.

**Comment by danoc** [comment]
Closing this one as part of a tidying up of our issue and PR backlog. Let's re-open once we can switch the base branch to `master` and rebase.

**Pull request closed without merging**